src/macros.jl

```diff
@@ -419,3 +419,47 @@ macro wrap(ex)
        Base.@__doc__ $(Expr(:call, name, lhsargs...)) = $body
     end)
 end
+
+"""
+    @install
+
+When applied to a unary method definition for the function `GapObj`,
+with argument of type `T`,
+this macro installs instead a three argument method for
+`GAP.julia_to_gap_internal`, with second argument of type
+`GAP.GapCacheDict` and third argument of type `Bool`.
+
+This way, the intended `GapObj(x::T)` method becomes available,
+and additionally its code is applicable in recursive calls,
+for example when `GapObj` is called with a vector of objects of type `T`.
+
+The calls of the macro have the form `GAP.@install GapObj(x::T) = f(x)`
+or `GAP.@install function GapObj(x::T) ... end`.
+"""
+macro install(ex)
+    errmsg = "GAP.@install must be applied to a unary method definition for GapObj"
+
+    # split the method definition
+    def_dict = try
+        MacroTools.splitdef(ex)
+    catch
+        error(errmsg)
+    end
+    def_dict[:name] === :GapObj || throw(ArgumentError(errmsg))
```

Comment:
What if a user doesn't want to import `GapObj` and does
```julia
GAP.@install GAP.GapObj(x::T) = f(x)
```
Does that work as well? If not, can we make it work with reasonable effort?

Reply:
O.k.